### PR TITLE
[master] 3.3.5 -> master cherry-pick

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  
-  
-  

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: src/server/game/Spells/SpellInfo.cpp [hunk 1]
```diff
@@ -1675,7 +1675,7 @@ bool SpellInfo::IsChanneled() const
 
 bool SpellInfo::IsMoveAllowedChannel() const
 {
-    return IsChanneled() && HasAttribute(SPELL_ATTR5_CAN_CHANNEL_WHEN_MOVING);
+    return IsChanneled() && (HasAttribute(SPELL_ATTR5_CAN_CHANNEL_WHEN_MOVING) || !HasChannelInterruptFlag(AURA_INTERRUPT_FLAG_MOVE | AURA_INTERRUPT_FLAG_TURNING));
```
Comment (funjoker) [inline src/server/game/Spells/SpellInfo.cpp:1678]: Do i have to cast to ChannelInterruptFlag here?
Or do i have to add HasChannelInterruptFlag which accepts AuraInterruptFlags ?

See original commit

Comment (Shauren) [reply src/server/game/Spells/SpellInfo.cpp:1678]: Just access ChannelInterruptFlags[0] directly

**Comment by funjoker** [comment]
April 2017 done. (some missing commits, which @Shauren needs to take a look at).

Missing:
2a462e93093d177a0e0bedf3e94ac2cf4e66a3c3
164af391c9662732dfc9cfe677c48bd3f927c2be
70c4a37544ade558fb79383d2668f5f0322d47b8
21b2042840e05ebeba365c54d47c07c13fde123f
2a04599ed8f471e584b3e3f72b210a67a4f1ab1c
25a449a90e01d7100d482c156fffa3ac2e7730d2
a4aa95a5a39d75b9e38fe03aac57ceb7d9bba90f
aa86dfddca8f1fcb6b77fc1a37f367978c593018
3ea46e57afe26778704647084cc6fa55a798e510

**Comment by funjoker** [comment]
April 2017 done. (some missing commits, which @Shauren needs to take a look at).

Missing:
2a462e93093d177a0e0bedf3e94ac2cf4e66a3c3
164af391c9662732dfc9cfe677c48bd3f927c2be
70c4a37544ade558fb79383d2668f5f0322d47b8
21b2042840e05ebeba365c54d47c07c13fde123f
2a04599ed8f471e584b3e3f72b210a67a4f1ab1c
25a449a90e01d7100d482c156fffa3ac2e7730d2
a4aa95a5a39d75b9e38fe03aac57ceb7d9bba90f
aa86dfddca8f1fcb6b77fc1a37f367978c593018
553103963fb7d027ede938c2e9ad9dccf6d7fd2c

**Comment by funjoker** [comment]
April 2017 done. (some missing commits, which @Shauren needs to take a look at).

Missing:
2a462e93093d177a0e0bedf3e94ac2cf4e66a3c3
164af391c9662732dfc9cfe677c48bd3f927c2be
70c4a37544ade558fb79383d2668f5f0322d47b8
21b2042840e05ebeba365c54d47c07c13fde123f
2a04599ed8f471e584b3e3f72b210a67a4f1ab1c
25a449a90e01d7100d482c156fffa3ac2e7730d2
a4aa95a5a39d75b9e38fe03aac57ceb7d9bba90f
aa86dfddca8f1fcb6b77fc1a37f367978c593018
553103963fb7d027ede938c2e9ad9dccf6d7fd2c

**Pull request merged**